test/cases/schema_dumper_test.rb

```diff
@@ -0,0 +1,262 @@
+          # but that doesn't work here because the schema dumper is not aware of which migration
+          # a column was added in.
+          assert output.include?('t.timestamp "this_should_change_to_timestamp"')
+          assert output.include?('t.timestamp "this_should_stay_as_timestamp"')
+        end
+      ensure
+        migration.migrate(:down)
+        # $stdout = original
+      end
+
+      def test_timestamps_schema_dump_before_rails_7_with_timestamptz_setting
+        migration, original, $stdout = nil, $stdout, StringIO.new
+
+        with_cockroachdb_datetime_type(:timestamptz) do
+          migration = Class.new(ActiveRecord::Migration[6.1]) do
+            def up
+              create_table("timestamps") do |t|
+                t.datetime :this_should_change_to_timestamp
+                t.timestamp :this_should_stay_as_timestamp
+                t.column :this_should_also_stay_as_timestamp, :timestamp
+              end
+            end
+            def down
+              drop_table("timestamps")
+            end
+          end
+          migration.migrate(:up)
+
+          output = perform_schema_dump
+          # Normally we'd write `t.datetime` here. But because you've changed the `datetime_type`
+          # to something else, `t.datetime` now means `:timestamptz`. To ensure that old columns
+          # are still created as a `:timestamp` we need to change what is written to the schema dump.
+          #
+          # Typically in Rails we handle this through Migration versioning (`ActiveRecord::Migration::Compatibility`)
+          # but that doesn't work here because the schema dumper is not aware of which migration
+          # a column was added in.
+
+          assert output.include?('t.timestamp "this_should_change_to_timestamp"')
+          assert output.include?('t.timestamp "this_should_stay_as_timestamp"')
+          assert output.include?('t.timestamp "this_should_also_stay_as_timestamp"')
+        end
+      ensure
+        migration.migrate(:down)
+        # $stdout = original
+      end
+
+      def test_schema_dump_with_correct_timestamp_types_via_add_column_before_rails_7_with_timestamptz_setting
+        migration, original, $stdout = nil, $stdout, StringIO.new
+
+        with_cockroachdb_datetime_type(:timestamptz) do
+          migration = Class.new(ActiveRecord::Migration[6.1]) do
+            def up
+              create_table("timestamps")
+
+              add_column :timestamps, :this_should_change_to_timestamp, :datetime
+              add_column :timestamps, :this_should_stay_as_timestamp, :timestamp
+            end
+            def down
+              drop_table("timestamps")
+            end
+          end
+          migration.migrate(:up)
+
+          output = perform_schema_dump
+          # Normally we'd write `t.datetime` here. But because you've changed the `datetime_type`
+          # to something else, `t.datetime` now means `:timestamptz`. To ensure that old columns
```

Comment:
These comments are from the original Active Record tests written in here: https://github.com/rails/rails/blob/main/activerecord/test/cases/schema_dumper_test.rb

I kept the comments because I saw that other original comments, from the other tests that inherit/override the original Active Record code, were also being kept.